### PR TITLE
Feature/expense chart by category

### PR DESCRIPTION
**Pull request by E-LeeSchaffer**

### CODE REVIEW

File: pages/report/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import dynamic from "next/dynamic";
+import Image from "next/image";
+import Link from "next/link";
+import styled from "styled-components";
+const BarChartPage = dynamic(() => import("@/components/Chart"), {
```
Comment (Leon910) [inline pages/report/index.js:5]: Her you are importing your bar component so I assume if under certain conditions the loading takes some time maybe you would want t include a loading message.

```suggestion
const BarChartPage = dynamic(() => import("@/components/Chart"), {
ssr: false,
  loading: () => <p>Loading chart...</p>
```

**Review by Leon910 — APPROVED** [review]
Hey @E-LeeSchaffer I didn´t find any major issues. You might want to include some checks (IsLoading, error) but for the functonality of your chart it shouldn´t cause any problems so far.

File: components/Chart.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { BarChart, YAxis, XAxis, Bar } from "recharts";
+import useSWR from "swr";
+
+export default function BarChartPage() {
+  const { data: transactionsList = [] } = useSWR("/api/transactions");
```
[the remark this answers is not in the document]
Comment (E-LeeSchaffer) [reply components/Chart.js:7]: Yes, thanks! We will include it overall in our code when we do our refactoring.

File: components/Chart.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import React, { useState } from "react";
```
[the remark this answers is not in the document]
Comment (E-LeeSchaffer) [reply components/Chart.js:1]: Indeed I don't need that, I have deleted it. Thanks!

File: components/Chart.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        onClick={() => handleCategoryClick(payload.value)}
+      >
+        {payload.value}
+      </text>
+    );
+  };
+
+  return (
+    <>
+      <Card>
```
[the remark this answers is not in the document]
Comment (E-LeeSchaffer) [reply components/Chart.js:64]: Is renamed.

**Pull request merged**